### PR TITLE
kernel config: Add back FTRACE_SYSCALLS.

### PR DESCRIPTION
**Pull request by eiffel-fl**
Hi.

###### Merge Checklist
**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

This PR adds back `CONFIG_FTRACE_SYSCALLS` to amd64 kernel.
This kernel option is needed by eBPF syscall tracer like [execsnoop](https://github.com/iovisor/bcc/blob/13b5563c11f7722a61a17c6ca0a1a387d2fa7788/libbpf-tools/execsnoop.bpf.c#L38) which we use in [inspektor-gadget](https://github.com/kinvolk/inspektor-gadget/blob/b49cea1baac0d7a5959b843d153728d448056f2f/pkg/gadgets/execsnoop/tracer/core/bpf/execsnoop.bpf.c#L38).

###### Change Log 
- Add back CONFIG_FTRACE_SYSCALLS to amd64 kernel.

###### Does this affect the toolchain?
**NO**

###### Test Methodology

I tested this option by building kernel package and installing on Azure Mariner 2.0.
Regarding boot time, there is almost no overhead with this option:

```bash
# Default kernel
francis-mariner-@francis-mariner-vm [ ~ ]$ uname -a
Linux francis-mariner-vm 5.15.41.1-1.cm2 #1 SMP Sat May 28 19:48:23 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
francis-mariner-@francis-mariner-vm [ ~ ]$ zgrep FTRACE_SYSCALLS /proc/config.gz
# CONFIG_FTRACE_SYSCALLS is not set
francis-mariner-@francis-mariner-vm [ ~ ]$ systemd-analyze
Startup finished in 4.815s (kernel) + 614ms (initrd) + 5.169s (userspace) = 10.599s
multi-user.target reached after 4.964s in userspace
# Custom kernel
francis-mariner-@francis-mariner-vm [ ~ ]$ uname -a 
Linux francis-mariner-vm 5.15.45.1-2.cm2 #1 SMP Tue Jun 21 13:58:40 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
francis-mariner-@francis-mariner-vm [ ~ ]$ zgrep FTRACE_SYSCALLS /proc/config.gz 
CONFIG_FTRACE_SYSCALLS=y
francis-mariner-@francis-mariner-vm [ ~ ]$ systemd-analyze
Startup finished in 4.802s (kernel) + 681ms (initrd) + 5.485s (userspace) = 10.970s 
multi-user.target reached after 5.480s in userspace
```

These numbers were also confirmed by Suresh Babu Chalamalasetty which I warmly thank for his help.

Best regards.

### CODE REVIEW

**Review by neha170 — COMMENTED** [review]
Hi Francis, thanks for making the changes and testing. Could you make the required changes in other kernel files so this gets reflected in the latest kernel build. You can use this PR as a sample. https://github.com/microsoft/CBL-Mariner/pull/3080. Please note that you would not need to make any changes in config_aarch64 file, but the rest of the files would have to be changes, as given in the sample PR.

**Comment by christopherco** [comment]
Hi Francis, I believe the issue here is actually triggered from moving to the `main` branch, which is indeed the correct branch to target for your PR.

> 
> I am indeed missing some packages which are needed by `srpmpacker` but I do not find how to get those:
> 
> 1. I do not understand why it does not get the packages as Makefile dependencies.

The packages listed here are not needed to run `srpmpacker`. Instead, `srpmpacker` is being run to create the srpms that the tools think need to be built as part of the kernel build (for example, buildrequires for the kernel spec). As for why this is happening, my guess here is that the branch switch to our `main` development branch may have touched the individual spec files, possibly bumping the versions, which leads `make` to believe that the dependency package needs to be rebuilt since the inputs could have changed.

> 2. I did not find in the documentation how to """hydrate""" the SRPM.

"Hydrating" the SRPM means downloading the source tarball so the SRPM can be created. The error is indicating that the tools could not find the corresponding source tarball. Agreed that "hydrating" is a bit of a confusing term. Will look to clarify this in the documentation.
We have a source server that we use to pull these sources tars. Please add the following option to your make command `SOURCE_URL=https://cblmarinerstorage.blob.core.windows.net/sources/core`

Also, I would recommend rebasing to the latest `main` to pick up the updated changes to the kernel specs. With how .specs have their own versioning/changelog, it can often times create merge conflicts when multiple spec file updates occur.

**Comment by christopherco** [comment]
> Hi.
> 
> I am not sure to understand but this command:
> 
> ```shell
> sudo make build-packages -j$(nproc) PACKAGE_BUILD_LIST="kernel" PACKAGE_REBUILD_LIST="kernel" CONFIG_FILE= REBUILD_TOOLS=y REBUILD_TOOLCHAIN=y REPO_LIST= REBUILD_DEP_CHAINS=n SOURCE_URL=https://cblmarinerstorage.blob.core.windows.net/sources/core
> ```
> 
> is building a lot of `.src.rpm` packages while I am just interested in one package: the kernel. Is there a way to skip the build of `*.src.rpm`? I checked the documentation but did not find anything at this subject.
> 
> Best regards and thank you in advance.

There are a few reasons why it is building various `.src.rpm` packages outside of the desired package (in this case the kernel package).
1. Unlike our releases for 1.0 and 2.0, we do not publish packages officially on packages.microsoft.com for the `main` development branch. And in order to build the desired package, one needs to have the package dependencies available and built. Therefore, in this case, the tools will build the package dependencies for the kernel package from source to make them available.
2. The 2.0 build system supports automatic/dynamic dependencies that are common in Fedora specs. The implication of this means that at the start of a build, there may not be a clear path to the goal node in the dependency graph, but as packages are built, these packages will produce implicit provides, which will then satisfy other package's dependencies, and eventually we get to a solved graph that can build the desired package. The scheduler will build packages it knows it can build to generate those implicit provides, thus leading to additional packages built outside of the expected/shortest path to the desired package.

We are developing ways to speed up / enlighten this build process, but as it stands right now, those enhancements are not available yet and so there likely will be additional packages built outside of the desired package.

**Comment by christopherco** [comment]
> Thank you for your detailled comment, it makes sense. Basically, Mariner build process (_i.e._ the scheduler) is grabbing the source to be able to build tools we need to build the kernel (_e.g._ `Makefile`).
> 
> Nonetheless, I am not sure to fully understand how the `scheduler` works. Indeed, why are these packages needed to build the kernel?
> 
> ```
> INFO[0024] 2 currently active build(s): [libcap-2.60-1.cm2.src.rpm libunistring-0.9.10-5.cm2.src.rpm]. 
> INFO[0024] Prebuilt: libcap-2.60-1.cm2.src.rpm -> [/home/francis-bake-mariner/CBL-Mariner/out/RPMS/x86_64/libcap-2.60-1.cm2.x86_64.rpm /home/francis-bake-mariner/CBL-Mariner/out/RPMS/x86_64/libcap-debuginfo-2.60-1.cm2.x86_64.rpm /home/francis-bake-mariner/CBL-Mariner/out/RPMS/x86_64/libcap-devel-2.60-1.cm2.x86_64.rpm]
> ```
The extra packages being built comes from the 2nd point about handling dynamic dependencies (see https://docs.fedoraproject.org/en-US/packaging-guidelines/PkgConfigBuildRequires/ for addition info on what we term as dynamic dependencies).

In addition, these dynamic dependencies are sometimes satisfied by RPM automatic provides. This provides data is only available once rpmbuild actually finishes building the package, since the data is automatically generated based on what is actually built. As a result, the scheduler will tend to build more than what is necessary for a specific package to ensure build correctness.

There is definitely room for improvement to optimize the dependency graph and enlighten the scheduler so the end build order execution is more optimal and we spend less time building packages we don't need to build, but currently we build more than necessary to ensure correctness of the requested package build.

**Review by christopherco — APPROVED** [review]
LGTM

File: SPECS-SIGNED/kernel-signed/kernel-signed.spec [hunk 1]
```diff
@@ -153,6 +153,9 @@ ln -sf linux-%{uname_r}.cfg /boot/mariner.cfg
 %exclude /module_info.ld
 
 %changelog
+* Fri Jul 08 2022 Francis Laniel <flaniel@linux.microsoft.com> - 5.15.45.1-3
```
Comment (christopherco) [inline SPECS-SIGNED/kernel-signed/kernel-signed.spec:156]:
```suggestion
* Fri Jul 08 2022 Francis Laniel <flaniel@linux.microsoft.com> - 5.15.48.1-5
```

File: SPECS/kernel-headers/kernel-headers.spec [hunk 2]
```diff
@@ -36,6 +36,9 @@ cp -rv usr/include/* /%{buildroot}%{_includedir}
 %{_includedir}/*
 
 %changelog
+* Fri Jul 08 2022 Francis Laniel <flaniel@linux.microsoft.com> - 5.15.45.1-3
```
Comment (christopherco) [inline SPECS/kernel-headers/kernel-headers.spec:39]:
```suggestion
* Fri Jul 08 2022 Francis Laniel <flaniel@linux.microsoft.com> - 5.15.48.1-5
```

File: SPECS/kernel/kernel.spec [hunk 3]
```diff
@@ -370,6 +370,10 @@ ln -sf linux-%{uname_r}.cfg /boot/mariner.cfg
 %{_sysconfdir}/bash_completion.d/bpftool
 
 %changelog
+* Fri Jul 08 2022 Francis Laniel <flaniel@linux.microsoft.com> - 5.15.45.1-3
```
Comment (christopherco) [inline SPECS/kernel/kernel.spec:373]:
```suggestion
* Fri Jul 08 2022 Francis Laniel <flaniel@linux.microsoft.com> - 5.15.48.1-5
```

**Review by christopherco — CHANGES_REQUESTED** [review]
Whoops, spoke a little too soon. Few corrections for the changelog

**Review by christopherco — APPROVED** [review]
:shipit:

**Review by rlmenge — APPROVED** [review]

**Review by neha170 — APPROVED** [review]

**Pull request merged**